straxen/plugins/event_processing.py

```diff
@@ -193,26 +164,81 @@ def infer_dtype(self):
             (f'alt_s2_x', np.float32,
              f'Alternate S2 reconstructed X position, uncorrected [cm]'),
             (f'alt_s2_y', np.float32,
-             f'Alternate S2 reconstructed Y position, uncorrected [cm]')]
-
-        # area before main S2
-        dtype += [
+             f'Alternate S2 reconstructed Y position, uncorrected [cm]'),
             (f'area_before_main_s2', np.float32,
              f'Sum of areas before Main S2 [PE]'),
             (f'large_s2_before_main_s2', np.float32,
-             f'The largest S2 before the Main S2 [PE]')]
+             f'The largest S2 before the Main S2 [PE]')
+        ]
+
+        dtype += self._get_posrec_dtypes()
+        return dtype
+
+    def _set_dtype_requirements(self):
+        """Needs to be run before inferring dtype as it is needed there"""
+        # Properties to store for each peak (main and alternate S1 and S2)
+        self.peak_properties = (
+            # name                dtype       comment
+            ('time',              np.int64,   'start time since unix epoch [ns]'),
+            ('center_time',       np.int64,   'weighted center time since unix epoch [ns]'),
+            ('endtime',           np.int64,   'end time since unix epoch [ns]'),
+            ('area',              np.float32, 'area, uncorrected [PE]'),
+            ('n_channels',        np.int32,   'count of contributing PMTs'),
+            ('n_competing',       np.float32, 'number of competing PMTs'),
+            ('max_pmt',           np.int16,   'PMT number which contributes the most PE'),
+            ('max_pmt_area',      np.float32, 'area in the largest-contributing PMT (PE)'),
+            ('range_50p_area',    np.float32, 'width, 50% area [ns]'),
+            ('range_90p_area',    np.float32, 'width, 90% area [ns]'),
+            ('rise_time',         np.float32, 'time between 10% and 50% area quantiles [ns]'),
+            ('area_fraction_top', np.float32, 'fraction of area seen by the top PMT array')
+        )
+
+    @staticmethod
+    def _get_si_dtypes(peak_properties):
+        """Get properties for S1/S2 from peaks directly"""
+        si_dtype = []
+        for s_i in [1, 2]:
+            # Peak indices
+            si_dtype += [
+                (f's{s_i}_index', np.int32, f'Main S{s_i} peak index in event'),
+                (f'alt_s{s_i}_index', np.int32, f'Alternate S{s_i} peak index in event')]
```

Comment:
Just noticed it, ,aybe we can change this to int16 or an uint16?

Reply:
Int16 could do yeah! Its not very heavy on the memort but yeah if we need an int32 for the number of peaks in one event, something is wrong

Reply:
okay, I thought about it a little more, but if we have a lot of noise e.g. due to a 1 PMT requirement, we don't want to fail because there are 32k peaks in one event. It's unlikely but not excluded.

Reply:
I did find some other funky choices that I will push

Reply:
```python
-            ('n_channels',        np.int32,   'count of contributing PMTs'),
-            ('n_competing',       np.float32, 'number of competing PMTs'),
+            ('n_channels',        np.int16,   'count of contributing PMTs'),
+            ('n_competing',       np.int32,   'number of competing peaks'),
```
this is the change I did make